File: 238-product-of-array-except-self/product-of-array-except-self.cpp
#include <vector>
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
       vector<int> left(nums.size());
       vector<int> right(nums.size());
       vector<int> answer(nums.size());
       left[0]=1;
       for (int i=1; i<left.size(); i++){
            left[i]=nums[i-1]*left[i-1];
       }    
       right[nums.size()-1]=1;
       for (int j=nums.size()-2; j>=0; j--){
            right[j]=nums[j+1]*right[j+1];
       }
       for(int k=0; k<nums.size(); k++){
            answer[k]=left[k]*right[k];
       }
       return answer;
    }
};

File: 238-product-of-array-except-self/product-of-array-except-self.py
class Solution(object):
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        result = [0]*len(nums)

        prefix_var = 1
        for i in range(len(nums)):
            result[i] = prefix_var
            prefix_var *= nums[i]

        postfix_var = 1
        for i in range(len(nums)-1, -1, -1):
            result[i] *= postfix_var
            postfix_var *= nums[i] 
        
        return result 